src/test/java/org/mule/tools/apikit/output/MuleConfigGeneratorTest.java

```diff
@@ -99,49 +108,138 @@ private GenerationModel mockGenerationModel(String example) {
+        assertEquals(expectedApiAutodiscoveryConfig, muleConfig.getApiAutodiscoveryConfig());
+        assertEquals(muleConfig.getHttpListenerConfigs().size(), 1);
+        assertEquals(muleConfig.getApikitConfigs().size(), 1);
+      } else {
+        assertEquals(muleConfig.getMainFlows().size(), 1);
+      }
+    }
+  }
+
   @Test
   public void blankDocumentWithoutLCInDomain() {
-    Document document = this.scaffoldBlankDocument(SHOW_CONSOLE);
-    Element rootElement = document.getRootElement();
-    assertCommonFlows(rootElement);
-    Element consoleFlow = rootElement.getChildren().get(3);
-    assertEquals("flow", consoleFlow.getName());
-    assertEquals("hello-console", consoleFlow.getAttribute("name").getValue());
-    assertEquals("httpListenerConfig", consoleFlow.getChildren().get(0).getAttribute("config-ref").getValue());
-    assertEquals("/console/*", consoleFlow.getChildren().get(0).getAttribute("path").getValue());
-    assertEquals("console", consoleFlow.getChildren().get(1).getName());
+    ScaffoldingConfiguration.Builder builder = ScaffoldingConfiguration.builder();
+    List<MuleConfig> muleConfigs = scaffoldBlankDocument(builder.build());
+    for (MuleConfig muleConfig : muleConfigs) {
+      Document document = muleConfig.getContentAsDocument();
+      Element rootElement = document.getRootElement();
+      assertCommonFlows(rootElement);
+      Element consoleFlow = rootElement.getChildren().get(3);
+      assertEquals("flow", consoleFlow.getName());
+      assertEquals("hello-console", consoleFlow.getAttribute("name").getValue());
+      assertEquals("httpListenerConfig", consoleFlow.getChildren().get(0).getAttribute("config-ref").getValue());
+      assertEquals("/console/*", consoleFlow.getChildren().get(0).getAttribute("path").getValue());
+      assertEquals("console", consoleFlow.getChildren().get(1).getName());
+    }
   }
 
   @Test
   public void blankDocumentWithoutLCInDomainHideConsole() {
-    Document document = scaffoldBlankDocument(HIDE_CONSOLE);
-    Element rootElement = document.getRootElement();
-    assertCommonFlows(rootElement);
-
+    ScaffoldingConfiguration.Builder builder = ScaffoldingConfiguration.builder();
+    List<MuleConfig> muleConfigs = scaffoldBlankDocument(builder.build());
+    for (MuleConfig muleConfig : muleConfigs) {
+      Document document = muleConfig.getContentAsDocument();
+      Element rootElement = document.getRootElement();
+      assertCommonFlows(rootElement);
+    }
   }
 
   private void assertCommonFlows(Element rootElement) {
     assertEquals("mule", rootElement.getName());
-    Element xmlListenerConfig = rootElement.getChildren().get(0);
-    assertEquals("listener-config", xmlListenerConfig.getName());
+    assertConfigurations(rootElement);
 
-    Element apikitConfig = rootElement.getChildren().get(1);
-    assertEquals("hello-config", apikitConfig.getAttribute("name").getValue());
+    assertMainFlow(rootElement, 2);
```

Comment:
Index not needed as parameter. `assertMainFlow` is only being called from here. If you want it to be reusable, send the child element instead.